src/test/CurveCallee.t.sol

```diff
@@ -0,0 +1,321 @@
+            address(this),
+            address(Chainlog(chainlog).getAddress("MCD_JOIN_ETH_A")),
+            uint256(0),
+            uint24(3000),
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        try Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        }) {
+            assertTrue(false);
+        } catch Error(string memory reason) {
+            assertEq(reason, "CurveCallee: only-wsteth");
+        }
+    }
+
+    function test_bigAmt() public {
+        uint256 amt = 5000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0),
+            uint24(3000),
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+    }
+
+    function test_profit() public {
+        uint256 minProfit = 10_000 * WAD;
+        uint256 amt = 50 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit),
+            uint24(3000),
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_poolFee() public {
+        uint24 poolFee = 500;
+        uint256 amt = 50 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0),
+            poolFee,
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+    }
+
+    function testFail_badPoolFee() public {
+        uint24 poolFee = 5000;
+        uint256 amt = 50 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0),
+            poolFee,
+            address(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
```

Comment:
does uniswap give a revert message that we could test here with the `try/catch` testing method?

Reply:
Unfortunately, no:

```haskell
      │  │  ├╴call 0xE592427A0AEce92De3Edee1F18E0157C05861564::0xc04b8d59000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000ce71065d4017f316ec606fe4422e11eb2c47c24600000000000000000000000000000000000000000000000000000000618c57fd0000000000000000000000000000000000000000000000026e65e90aec54b09c0000000000000000000000000000000000000000000023ff491dc445f5506a44000000000000000000000000000000000000000000000000000000000000002bc02aaa39b223fe8d0a0e5c4f27ead9083c756cc20013886b175474e89094c44da98b954eedeac495271d0f000000000000000000000000000000000000000000 (src/CurveCallee.sol:167)
      │  │  │  └╴error Revert 0x <no source map>
      │  │  └╴error Revert 0x (src/CurveCallee.sol:167)
      │  └╴error Revert 0x <no source map>
      └╴error Revert 0x (src/test/CurveCallee.t.sol:237)
```